core/src/test/java/org/apache/spark/launcher/SparkLauncherSuite.java

```diff
@@ -137,7 +139,9 @@ public void testInProcessLauncher() throws Exception {
       // Here DAGScheduler is stopped, while SparkContext.clearActiveContext may not be called yet.
       // Wait for a reasonable amount of time to avoid creating two active SparkContext in JVM.
```

Comment:
why is `500 ms` not a reasonable waiting time anymore?

Reply:
Here Vanzin is waiting for the active SparkContext being empty.

Reply:
It's not that 500ms is not reasonable, it's that this code will return more quickly when the context shuts down in under 500ms, have a longer grace period in case it ever takes more than 500ms, and fail the test here if the context does not shut down.